src/main/java/roomescape/service/ReservationService.java

```diff
@@ -87,26 +108,71 @@ private void validatePastDate(final ReservationRequest reservationRequest, final
         }
     }
 
+    @Transactional
     public void delete(final Long id) {
-        reservationRepository.deleteById(id);
+        final Optional<Reservation> reservationResult = reservationRepository.findById(id);
+        if (reservationResult.isPresent() && reservationResult.get().getStatus() == ReservationStatus.RESERVED) {
+            updateWaitReservation(reservationResult.get());
+            reservationRepository.deleteById(id);
+        }
```

Comment:
```suggestion
        reservationRepository.findById(id)
                .filter(reservation -> reservation.getStatus() == ReservationStatus.RESERVED)
                .ifPresent(reservation -> {
                    updateWaitReservation(reservation);
                    reservationRepository.deleteById(id);
                });
```

`Optional`에서 `isPresent()`로 확인후 `get()`을 하신것으로 보이는데요. `Optional`에서 `get()`은 체크를 누락하는 등 결국 문제가 발생할 수 있기 때문에 지양하는 것이 좋습니다. 당초에 목적 자체가 not-null 케이스에 대해서 별도로 분리해서 사용하기보다는 한번에 없는 케이스에 대한 고려를 함께하여야해요.

Reply:
감사합니다 optional도 filter를 쓸 수 있었군요 👀